### ISSUE TITLE
bug: The iam policy can't be created in AWS China

### ISSUE THREAD

**Issue by Wyifei** [lablabs/terraform-aws-eks-external-dns#31]
### Summary

The iam policy can't be created in AWS China, because AWS china the arn format is: arn:awscn

### Issue Type

Bug Report

### Terraform Version

```console
terraform --version
Terraform v1.3.6
on linux_amd64
+ provider registry.terraform.io/cloudposse/utils v1.6.0
+ provider registry.terraform.io/hashicorp/aws v4.45.0
+ provider registry.terraform.io/hashicorp/external v2.2.3
+ provider registry.terraform.io/hashicorp/helm v2.7.1
+ provider registry.terraform.io/hashicorp/kubernetes v2.11.0
```

### Steps to Reproduce

module "external_dns" {
  count   = var.environment == "dev" ? 1 : 0
  source  = "lablabs/eks-external-dns/aws"
  version = "1.1.0"

  cluster_identity_oidc_issuer     = data.aws_iam_openid_connect_provider.meid_eks_oidc.url
  cluster_identity_oidc_issuer_arn = data.aws_iam_openid_connect_provider.meid_eks_oidc.arn

  irsa_role_name_prefix = "irsa-external-dns-${var.projectname}-${var.environment}-${var.region}"
  irsa_tags = {
    Name        = "external-dns-${var.projectname}-${var.environment}-${var.region}"
    Region      = var.region
    Environment = var.environment
    Managedby   = "Terraform"
  }

  helm_chart_version = "6.8.1"

  values = yamlencode({
    policy             = "sync"
    triggerLoopOnEvent = true
    txtPrefix          = "z-external-dns-"
    global             = { imageRegistry = "${var.accountid}.dkr.ecr.${var.region}.${var.awsdomain}" }
  })
}

### Expected Results

The iam policy cloud be created successfully

### Actual Results

```console
Error: error creating IAM Policy irsa-external-dns-meid-cn-dev-cn-north-1-external-dns: MalformedPolicyDocument: Partition "aws" is not valid for resource "arn:aws:route53:::hostedzone/*".
│       status code: 400, request id: ecc835f4-88df-401b-a4a5-36a16e692e8b
│ 
│   with module.external_dns[0].aws_iam_policy.this[0],
│   on .terraform/modules/external_dns/iam.tf line 49, in resource "aws_iam_policy" "this":
│   49: resource "aws_iam_policy" "this" {
```

**Comment by Wyifei** [comment]
Looking forward version update

**Comment by Wyifei** [comment]
=.=

**Comment by tomas-balaz** [comment]
Hi @Wyifei 

We've just released the new version of this module ([v1.2.0](https://github.com/lablabs/terraform-aws-eks-external-dns/releases/tag/v1.2.0))

You can use `aws_partition` variable now, as you can see in https://github.com/lablabs/terraform-aws-eks-external-dns/pull/35#issuecomment-1611780101
